### PR TITLE
Service Account Provider - Workload Identity Federation workaround for Gitlab CI

### PR DESCRIPTION
**Pull request by cgrotz**
This is the source code for the Service Account Provider, a workaround for using GitLab CI together with Google Cloud APIs, to workaround some gaps in Workload identity federations OpenID Connect support.

### CODE REVIEW

File: helpers/exclusion_list.txt [hunk 1]
```diff
@@ -23,6 +23,7 @@
 ./tools/ml-dataprep
 ./tools/ml-auto-eda
 ./tools/netblock-monitor
+./tools/service-account-provider/example_gitlab_pipeline
```
Comment (boredabdel) [inline helpers/exclusion_list.txt:26]: Why did you add this folder to the exclusion list ? Is the linting not passing ?

File: tools/service-account-provider/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+WORKDIR /go/src/app
+ADD . /go/src/app
+
+RUN go build -o /go/bin/app main.go
+
+FROM gcr.io/distroless/base-debian10
+COPY --from=build /etc/ssl/certs /etc/ssl/certs
+COPY --from=build /go/bin/app /
+
+CMD ["/app"]
```
Comment (boredabdel) [inline tools/service-account-provider/Dockerfile:26]: Please use a multi-stage docker file ? https://docs.docker.com/develop/develop-images/multistage-build/

Essentially build using the golang image but the final image should not contain the compiler, you can use Alpine or Debian

File: tools/service-account-provider/README.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
+# Service Account Provider (SAPRO)
```
Comment (boredabdel) [inline tools/service-account-provider/README.md:1]: The Readme is missing instructions on how to setup and use the tool

**Review by boredabdel — CHANGES_REQUESTED** [review]
Few comments

File: helpers/exclusion_list.txt (hunk 1 above)
Comment (cgrotz) [reply helpers/exclusion_list.txt:26]: There are a few environment variables that are not set within the shell script (they are set from within the GitLab pipeline), and the linter was complaining about them. Since the script is just an example for how to use the tool, I added the exclusion.
Although it would be great if you have a suggestion how I could handle that differently.

File: tools/service-account-provider/Dockerfile (hunk 2 above)
Comment (cgrotz) [reply tools/service-account-provider/Dockerfile:26]: I am sorry, I thought I am using a multi-stage. Can you take another look, I am not sure what I am missing.

File: tools/service-account-provider/README.md (hunk 3 above)
Comment (cgrotz) [reply tools/service-account-provider/README.md:1]: I added some instructions. Thanks for pointing this out.

**Review by boredabdel — APPROVED** [review]
LGTM
